integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/schema/management/ElasticsearchIndexSchemaManagerExporterIT.java

```diff
@@ -0,0 +1,134 @@
+
+	@Before
+	public void setUp() {
+		setupHelper.start()
+				.withBackendProperty(
+						ElasticsearchIndexSettings.ANALYSIS_CONFIGURER,
+						// use this configurer to produce a huge config:
+						new ElasticsearchIndexSchemaManagerAnalyzerITAnalysisConfigurer()
+				)
+				.withIndex( mainIndex )
+				.setup();
+	}
+
+	@Test
+	public void export() throws IOException {
+		Path directory = temporaryFolder.newFolder().toPath();
+		String testIndexName = "test";
+		mainIndex.schemaManager().exportExpectedSchema( new SearchSchemaCollector() {
+			@Override
+			public void indexSchema(Optional<String> backendName, String indexName, SchemaExport export) {
+				export.toFiles( directory.resolve( testIndexName ) );
+			}
+		} );
+
+		assertJsonEquals(
+				"{" +
+						"  \"aliases\": {" +
+						"    \"main-write\": {" +
+						"      \"is_write_index\": true" +
+						"    }," +
+						"    \"main-read\": {" +
+						"      \"is_write_index\": false" +
+						"    }" +
+						"  }," +
+						"  \"mappings\": {" +
+						"    \"properties\": {" +
+						"      \"_entity_type\": {" +
+						"        \"type\": \"keyword\"," +
+						"        \"index\": false," +
+						"        \"doc_values\": true" +
+						"      }" +
+						"    }," +
+						"    \"dynamic\": \"strict\"" +
+						"  }," +
+						"  \"settings\": {" +
+						"    \"analysis\": {" +
+						"      \"analyzer\": {" +
+						"        \"custom-analyzer\": {" +
+						"          \"type\": \"custom\"," +
+						"          \"tokenizer\": \"custom-edgeNGram\"," +
+						"          \"filter\": [" +
+						"            \"custom-keep-types\"," +
+						"            \"custom-word-delimiter\"" +
+						"          ]," +
+						"          \"char_filter\": [" +
+						"            \"custom-pattern-replace\"" +
+						"          ]" +
+						"        }" +
+						"      }," +
+						"      \"tokenizer\": {" +
+						"        \"custom-edgeNGram\": {" +
+						"          \"type\": \"edge_ngram\"," +
+						"          \"min_gram\": 1," +
+						"          \"max_gram\": 10" +
+						"        }" +
+						"      }," +
+						"      \"filter\": {" +
+						"        \"custom-keep-types\": {" +
+						"          \"type\": \"keep_types\"," +
+						"          \"types\": [" +
+						"            \"\\u003cNUM\\u003e\"," +
+						"            \"\\u003cDOUBLE\\u003e\"" +
+						"          ]" +
+						"        }," +
+						"        \"custom-word-delimiter\": {" +
+						"          \"type\": \"word_delimiter\"," +
+						"          \"generate_word_parts\": false" +
+						"        }" +
+						"      }," +
+						"      \"char_filter\": {" +
+						"        \"custom-pattern-replace\": {" +
+						"          \"type\": \"pattern_replace\"," +
+						"          \"pattern\": \"[^0-9]\"," +
+						"          \"replacement\": \"0\"," +
+						"          \"tags\": \"CASE_INSENSITIVE|COMMENTS\"" +
+						"        }" +
+						"      }" +
+						"    }" +
+						"  }" +
+						"}",
```

Comment:
We'll need to run those tests against all Elasticsearch/OpenSearch versions. I suspect you won't get the same JSON in all case and thus get failures...

It might be a good idea to be less strict in your assertions, i.e. use `assertJsonEqualsIgnoringUnknownFields` and only check for a few fields. Also, I'd disable this test for ES 6 and below, because it's going to fail on older versions for sure (because of the concept of named types).